it/controllers/incomeSources/add/AddBusinessTradeControllerISpec.scala

```diff
@@ -0,0 +1,177 @@
+  val addBusinessTradeSubmitUrl: String = controllers.incomeSources.add.routes.AddBusinessTradeController.submit(isAgent = false, isChange = false).url
+  val changeBusinessTradeUrl: String = controllers.incomeSources.add.routes.AddBusinessTradeController.show(isAgent = false, isChange = true).url
+  val submitChangeBusinessTradeUrl: String = controllers.incomeSources.add.routes.AddBusinessTradeController.submit(isAgent = false, isChange = true).url
+
+  val addBusinessAddressUrl: String = controllers.incomeSources.add.routes.AddBusinessAddressController.show(isChange = false).url
+  val incomeSourcesUrl: String = controllers.routes.HomeController.show().url
+  val checkDetailsUrl: String = controllers.incomeSources.add.routes.CheckBusinessDetailsController.show().url
+
+  val pageTitleMsgKey: String = messagesAPI("add-business-trade.heading")
+  val pageHint: String = messagesAPI("add-business-trade.p1")
+  val button: String = messagesAPI("base.continue")
+
+
+  s"calling GET $addBusinessTradeControllerShowUrl" should {
+    "render the Add Business trade page for an Individual" when {
+      "User is authorised" in {
+        Given("I wiremock stub a successful Income Source Details response with multiple businesses and a uk property")
+        enable(IncomeSources)
+        IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, multipleBusinessesResponse)
+        When(s"I call GET $addBusinessTradeControllerShowUrl")
+        val result = IncomeTaxViewChangeFrontend.getAddBusinessTrade
+        verifyIncomeSourceDetailsCall(testMtditid)
+
+        result should have(
+          httpStatus(OK),
+          pageTitleIndividual(pageTitleMsgKey),
+          elementTextByID("business-trade-hint")(pageHint),
+          elementTextByID("continue-button")(button)
+        )
+      }
+
+    }
+    "303 SEE_OTHER - redirect to home page" when {
+      "Income Sources FS disabled" in {
+
+        Given("I wiremock stub a successful Income Source Details response with no businesses or properties")
+        disable(IncomeSources)
+        IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, noPropertyOrBusinessResponse)
+
+        When(s"I call GET ${addBusinessTradeControllerShowUrl}")
+        val result = IncomeTaxViewChangeFrontend.getAddBusinessTrade
+
+        result should have(
+          httpStatus(SEE_OTHER),
+          redirectURI(incomeSourcesUrl)
+        )
+      }
+    }
+  }
+
+  s"calling POST ${addBusinessTradeSubmitUrl}" should {
+    s"303 SEE_OTHER and redirect to $addBusinessAddressUrl" when {
+      "User is authorised and business trade is valid" in {
+        enable(IncomeSources)
+        IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, noPropertyOrBusinessResponse)
+
+        val formData: Map[String, Seq[String]] = {
+          Map(
+            BusinessTradeForm.businessTrade -> Seq("Test Business Trade")
+          )
+        }
+
```

Comment:
Can you add a check for business trade name ? check with database using the session service

also you need to set the business name in session